### PR TITLE
Add JWT auth middleware, RLS policies, and CLI commands

### PR DESCRIPTION
**Pull request by cmeans**
## Summary
- **AuthMiddleware**: validates Bearer JWT tokens, extracts `owner_id` from `sub` claim, sets contextvar. Opt-in via `AWARENESS_AUTH_REQUIRED=true`. Skips auth for `/health` and `/favicon.ico`.
- **Row-level security**: Postgres RLS policies on all 4 data tables (`entries`, `reads`, `actions`, `embeddings`) as defense-in-depth. Uses `set_config('app.current_user', ...)` scoped per-transaction.
- **CLI: `mcp-awareness-user`**: add/list/set-password/export/delete users. Email canonical normalization (gmail dots, +tags), E.164 phone validation, argon2id password hashing.
- **CLI: `mcp-awareness-token`**: generate JWTs for self-hosted multi-user deployments.
- **CLI: `mcp-awareness-secret`**: generate 256-bit JWT signing secrets.
- **New dependencies**: `PyJWT`, `argon2-cffi`, `phonenumbers`
- Middleware order: AuthMiddleware (outermost) → SecretPathMiddleware/HealthMiddleware → FastMCP app
- Stdio transport always uses `DEFAULT_OWNER` — no auth check

**PR 3 of 3** for multi-tenant foundation (Phase 1). Depends on PRs #89 and #90 (both merged).

### Prerequisites
- `pip install -e ".[dev]"`
- Deploy to test instance on alternate port (`AWARENESS_PORT=8421`)

### Manual tests (via MCP tools + CLI)
1. - [x] **Auth disabled (default) — existing behavior preserved**
   With `AWARENESS_AUTH_REQUIRED` unset or `false`:
   ```
   get_briefing()
   ```
   Expected: works without any token, uses `DEFAULT_OWNER`

2. - [x] **Generate secret and token via CLI**
   ```bash
   mcp-awareness-secret
   export AWARENESS_JWT_SECRET=<output>
   mcp-awareness-token --user testuser --expires 1d
   ```
   Expected: secret is a 43-char base64url string, token is a valid JWT

3. - [x] **Auth enabled — valid token works**
   Set `AWARENESS_AUTH_REQUIRED=true` and `AWARENESS_JWT_SECRET`, restart server.
   Call with valid Bearer token:
   Expected: request succeeds, entries scoped to token's `sub` user

4. - [x] **Auth enabled — missing token rejected**
   Call without Authorization header:
   Expected: 401 with `{"error": "Missing or invalid Authorization header"}`

5. - [x] **User management CLI**
   ```bash
   mcp-awareness-user add alice --email alice+test@gmail.com --display-name "Alice"
   mcp-awareness-user list
   mcp-awareness-user delete alice --confirm
   ```
   Expected: user created (canonical_email = `alice@gmail.com`), listed, deleted

6. - [x] **All tests pass**
   `pytest tests/ -v` — 454 tests should pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by cmeans — COMMENTED** [review]
## QA Re-Review — PR #91 (Round 2)

Commit `c284252`: \`users.updated\` changed from \`NOT NULL DEFAULT now()\` to nullable (\`NULL\` until first update). Change applied consistently in migration, DDL (\`create_tables.sql\`), CLI (\`_user_add\`), and data dictionary.

- pytest: 450/450 pass
- CI: all green, codecov/patch SUCCESS
- No functional impact to auth, RLS, or CLI tests

All prior test results remain valid. Ready for QA Signoff.

**Comment by cmeans** [comment]
Adding Ready for QA Signoff — users.updated fix verified across migration, DDL, CLI, and docs. CI fully green, 450/450 pytest.

**Review by cmeans — COMMENTED** [review]
We need to support/enforce stronger passwords...their personal data should be properly protected.

File: src/mcp_awareness/cli.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+        cur.execute(
+            "SELECT * FROM reads WHERE owner_id = %s ORDER BY timestamp",
+            (args.user_id,),
+        )
+        reads = cur.fetchall()
+        for r in reads:
+            for k, v in r.items():
+                if isinstance(v, datetime):
+                    r[k] = v.isoformat()
+        data["reads"] = reads
```
Comment (cmeans) [inline src/mcp_awareness/cli.py]: Do we really want to share what is essentially an internal user-support table?  We can discuss.

**Review by cmeans — COMMENTED** [review]
We need to surface the password requirements for users.
I'm not sure if we should be exporting the reads table for users.  Let's talk.

File: src/mcp_awareness/cli.py [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+
+    result = zxcvbn(password, user_inputs=[args.user_id])
+    if result["score"] < 3:
+        warning = result["feedback"].get("warning", "")
+        suggestions = result["feedback"].get("suggestions", [])
+        print(f"Error: password too weak{': ' + warning if warning else ''}", file=sys.stderr)
+        for s in suggestions:
+            print(f"  - {s}", file=sys.stderr)
+        sys.exit(1)
+
```
Comment (cmeans) [inline src/mcp_awareness/cli.py]: Do this before you ask the user to enter a confirmation password, otherwise you're making them type crap in twice.  Not friendly.

**Review by cmeans — COMMENTED** [review]
Still a poor user experience.  Making the user enter a potentially bad password twice rather than short-circuiting the second entry when you should already know if it's good enough or not.

**Review by cmeans — COMMENTED** [review]
## QA Full Re-Review — PR #91 (Round 4)

Complete re-review from scratch covering all 8 commits, 13 changed files, 1930 insertions.

---

### Code Review

**AuthMiddleware** (middleware.py)
- JWT validation: Bearer token extraction, `jwt.decode()` with configurable algorithm
- Context management: `_owner_ctx.set(owner_id)` in try/finally with `reset()` — clean on exceptions
- Path exclusions: `/health` and `/favicon.ico` bypass auth
- Error handling: separate 401 messages for missing header, expired token, invalid token, missing `sub` claim
- Non-JWT exceptions re-raised (not swallowed) — verified by `test_unexpected_exception_re_raises`
- Opt-in: `AWARENESS_AUTH_REQUIRED=true`, fails fast with `ValueError` if `JWT_SECRET` not set

**RLS Policies** (g2b3c4d5e6f7_add_rls_policies.py)
- `ENABLE ROW LEVEL SECURITY` on entries, reads, actions, embeddings
- `owner_isolation` (USING) for SELECT/UPDATE/DELETE
- `owner_insert` (WITH CHECK) for INSERT
- Uses `current_setting('app.current_user', true)` — graceful empty string when unset
- Clean downgrade: drop policies + disable RLS
- Revision chain correct: depends on `f1a2b3c4d5e6`

**PostgresStore RLS Wiring** (postgres_store.py)
- `_set_rls_context(cur, owner_id)` calls `set_config('app.current_user', %s, true)` — transaction-local
- Every `with self._pool.connection()` now includes `conn.transaction()` before cursor
- Applied consistently across all ~30 methods — verified by inspection

**CLI** (cli.py)
- `secret_main()`: `secrets.token_urlsafe(32)` → 256-bit, 43-char base64url
- `token_main()`: JWT with sub/iat/exp, configurable expiry (d/h), reads algorithm from env
- `user_main()`: Full CRUD with subcommands (add/list/set-password/export/delete)
- `_canonical_email()`: lowercase, strip +tags, gmail dot-stripping, googlemail→gmail normalization
- `_validate_phone()`: E.164 via phonenumbers library, rejects invalid numbers
- `_user_set_password()`:
  - **Validate-before-confirm**: prompts password first, validates strength, then asks for confirmation
  - **Retry loop**: 3 attempts for both strength and mismatch failures
  - **Validation order**: max 128 chars (DoS) → min 14 chars → zxcvbn score >= 3
  - **Fallback message**: "Password is too easily guessed" when zxcvbn gives no feedback
  - **UX**: prints requirements upfront, lists all issues with bullet points, shows remaining attempts
- `_user_export()`: GDPR data export (user + entries + reads + actions) as JSON
- `_user_delete()`: hard delete with `--confirm` gate, cascading order (reads→actions→entries→users)

**DDL Consistency**
- `users.updated` is nullable everywhere: migration (`TIMESTAMPTZ`), `create_tables.sql`, test DDL, data-dictionary
- Verified all 4 touchpoints match

**Dependencies**
- `PyJWT>=2.8`, `argon2-cffi>=23.1`, `phonenumbers>=8.13`, `zxcvbn-python>=4.4`
- mypy overrides for jwt and zxcvbn

**Server Wiring**
- AuthMiddleware wraps outermost in both MOUNT_PATH and non-MOUNT_PATH HTTP paths
- Stdio transport unchanged (DEFAULT_OWNER, no auth)

### Observations

1. **`zxcvbn-python` emits DeprecationWarning on import** — the package itself warns "deprecated". The `zxcvbn` package (4.5.0) appears to be the actively maintained fork. Not a blocker — the functionality works — but consider switching the dependency.
2. **PR body test count was 453, actual is 454** — updated to match.

### CI Gate Check

| Check | Conclusion |
|-------|------------|
| lint | ✅ SUCCESS |
| typecheck | ✅ SUCCESS |
| test (3.10) | ✅ SUCCESS |
| test (3.11) | ✅ SUCCESS |
| test (3.12) | ✅ SUCCESS |
| **codecov/patch** | ✅ **SUCCESS** |

**Zero non-SUCCESS/non-SKIPPED checks.**

### Test Results

| Check | Result |
|-------|--------|
| pytest | ✅ 454/454 pass |
| ruff (src/, alembic/) | ✅ Clean |
| mypy | ✅ Clean |
| CI | ✅ All green |
| Manual #1: Auth disabled — default behavior | ✅ briefing works without token |
| Manual #2: Secret + token generation | ✅ 43-char secret, valid JWT with sub/iat/exp |
| Manual #3: Auth enabled — valid token | ✅ stats returned (total=17) |
| Manual #4: Auth enabled — missing token | ✅ 401 "Missing or invalid Authorization header" |
| Bonus: Expired token | ✅ 401 "Token expired" |
| Bonus: Wrong secret | ✅ 401 "Invalid token" |
| Manual #5: User CLI — add with email norm | ✅ Alice+Test@Gmail.com → canonical alice@gmail.com |
| Manual #5: User CLI — add with phone | ✅ +14155551234 stored in E.164 |
| Manual #5: User CLI — timezone | ✅ America/Chicago stored correctly |
| Manual #5: User CLI — updated is NULL | ✅ new users have updated=None |
| Manual #5: User CLI — export | ✅ GDPR export with user/entries/reads/actions |
| Manual #5: User CLI — delete | ✅ cascading hard delete with counts |
| Manual #6: pytest count | ✅ 454/454 (PR body updated) |

### Verdict

Zero findings (1 observation about zxcvbn-python deprecation). All CI green. All manual tests pass including bonus auth edge cases. Ready for QA Signoff.
